.azure-pipelines/release-dev.yml

```diff
@@ -0,0 +1,42 @@
+# Build the dev version of the package and publish to artifacts
+
+trigger: none
+
+pool:
+  vmImage: ubuntu-latest
+steps:
+  - displayName: 'Use Python 3.10'
+    task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.10'
+  - displayName: Install Python build dependencies
```

Comment:
Should we move the reusable bits into a common template? I don't know how to do that.

Reply:
Sounds good. That's doable in AzurePipelines: https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops&pivots=templates-includes.

A while ago it's not doable in GitHub Action, but it may now.

Reply:
I just want them to work before refining. Maybe merge the current thing and create the pipelines first?

Reply:
Yes. Let's forward this PR first